operations/helm/charts/grafana-agent/templates/hpa.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if and (eq .Values.controller.type "deployment") .Values.controller.autoscaling.enabled }}
```

Comment:
Would it be possible to use an HPA with a StatefulSet too?

Reply:
Yes, it is possible to use an HPA with a StatefulSet. But to reduce the nervousness about this change, it is only targeted toward deployment. If we are comfortable with adding HPA for StatefulSet, I can make the change to add it. Thanks!